apps/rush-lib/src/api/CommandLineConfiguration.ts

```diff
@@ -7,20 +7,87 @@ import { JsonFile, JsonSchema, FileSystem } from '@rushstack/node-core-library';
 
 import { RushConstants } from '../logic/RushConstants';
 
-import {
+import type {
   CommandJson,
   ICommandLineJson,
   IPhaseJson,
-  ParameterJson,
-  IPhasedCommandJson
+  IPhasedCommandJson,
+  IBulkCommandJson,
+  IGlobalCommandJson,
+  IFlagParameterJson,
+  IChoiceParameterJson,
+  IStringParameterJson
 } from './CommandLineJson';
+import type { RushConfigurationProject } from './RushConfigurationProject';
 
 const EXPECTED_PHASE_NAME_PREFIX: '_phase:' = '_phase:';
 
 export interface IShellCommandTokenContext {
   packageFolder: string;
 }
 
+export interface IPhase extends IPhaseJson {
+  isSynthetic?: boolean;
+  logFilenameIdentifier: string;
+  getDisplayNameForProject(rushProject: RushConfigurationProject): string;
+}
+
+export interface ICommandWithParameters {
+  associatedParameters: Set<Parameter>;
+}
+export interface IPhasedCommand extends IPhasedCommandJson, ICommandWithParameters {
+  isSynthetic?: boolean;
+  watchForChanges?: boolean;
+  disableBuildCache?: boolean;
```

Comment:
Should these be optional?  Optional parameters introduce additional states that complicate an algorithm; this can make code more readable, but it's only worthwhile if there are lots of callers.

Reply:
`watchForChanges` and `disableBuildCache` are optional in `IBulkCommandJson`, and the synthetic phased commands are generated from the bulk command, so I'd rather keep those optional, but I did change `isSynthetic` to be required.